src/popup/popup.js

```diff
@@ -286,18 +286,27 @@ elements.searchIcon.addEventListener('click', () => {
     });
   elements.clearSearchButton[0].classList.remove('display-none');
 });
-
+elements.modal.addEventListener('click', () => {
+  elements.modalCancel.click();
+});
 elements.clearSearchButton[0].addEventListener('click', () => {
-  // Restore Initial Content
-  elements.clearSearchButton[0].classList.add('display-none');
-  elements.inputField.value = '';
-  removeOldSearchResults();
-  removeLoadMoreButton(elements.loadMoreButtonWrapper);
-  elements.gridPrimary.setAttribute('style', 'position: relative; height: 0px;');
-  localStorage.clear();
-  restoreInitialContent('primary');
-
-  applyFilters();
+  const modalText = 'Do you really want to clear the search?';
+  function onModalConfirm() {
+    // Restore Initial Content
+    elements.clearSearchButton[0].classList.add('display-none');
+    elements.inputField.value = '';
+    removeOldSearchResults();
+    removeLoadMoreButton(elements.loadMoreButtonWrapper);
+    elements.gridPrimary.setAttribute('style', 'position: relative; height: 0px;');
+    localStorage.clear();
+    restoreInitialContent('primary');
+    applyFilters();
+    elements.modal.style.display = 'none';
+  }
+  function onModalClose() {
+    elements.modal.style.display = 'none';
```

Comment:
It would be good if you use `display-none` CSS utility defined in `_utilities.scss`.